src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/HostTests.cs

```diff
@@ -46,6 +48,45 @@ public void CreateDefaultBuilder_IncludesContentRootByDefault()
             Assert.Equal(expected, env.ContentRootPath);
         }
 
+        public static bool IsWindowsAndRemotExecutorIsSupported => PlatformDetection.IsWindows && RemoteExecutor.IsSupported;
+
+        [ConditionalFact(typeof(HostTests), nameof(IsWindowsAndRemotExecutorIsSupported))]
+        public void CreateDefaultBuilder_DoesNotChangeContentRootIfCurrentDirectoryIsWindowsSystemDirectory()
+        {
+            using var _ = RemoteExecutor.Invoke(() =>
+            {
+                string originalCwd = Environment.CurrentDirectory;
+                // Test that the path gets normalized before comparison. Use C:\WINDOWS\SYSTEM32\ instead of C:\Windows\system32.
+                string systemDirectory = Environment.GetFolderPath(Environment.SpecialFolder.System).ToUpper() + "\\";
+
+                try
+                {
+                    Environment.CurrentDirectory = systemDirectory;
+
+                    IHostBuilder builder = Host.CreateDefaultBuilder();
+                    using IHost host = builder.Build();
+
+                    var config = host.Services.GetRequiredService<IConfiguration>();
+                    var env = host.Services.GetRequiredService<IHostEnvironment>();
+
+                    // Temporary for diagnostics from Windows.Nano.1809.Amd64.Open containers on helix where config["ContentRoot"] is C:\.
+                    // Maybe this doesn't happen anymore if the container doesn't support RemoteExecutor.
+                    if (config["ContentRoot"] is not null)
+                    {
+                        throw new Exception($"ContentRoot: {config["ContentRoot"]}, originalCwd: {originalCwd}, systemDirectory: {systemDirectory}, Environment.CurrentDirectory: {Environment.CurrentDirectory}, Environment.SpecialFolder.System: {Environment.SpecialFolder.System}");
```

Comment:
```suggestion
                        throw new Exception($"ContentRoot: {config["ContentRoot"]}, originalCwd: {originalCwd}, systemDirectory: {systemDirectory}, Environment.CurrentDirectory: {Environment.CurrentDirectory}, Environment.SpecialFolder.System: {Environment.GetFolderPath(Environment.SpecialFolder.System)}");
```

Reply:
It looks like the Windows container doesn't have a system directory. I guess we could just skip this test if `string.IsNullOrEmpty(Environment.GetFolderPath(Environment.SpecialFolder.System))`. We can just add to the already-verbose skip condition: `IsWindowsAndHasSystemFolderAndRemotExecutorIsSupported`😆

Reply:
Looks like this is caused by https://github.com/dotnet/runtime/issues/21430. I disabled the test when `Environment.GetFolderPath(Environment.SpecialFolder.System)` returns null or empty.